### PR TITLE
Updated metaviper/gsea enrichment

### PR DESCRIPTION
**Pull request by allaway**
By drug, this time. This is more interesting than the previous one, but based on the network plots it's pretty clear that this analysis effectively disregards any drugs with only a few listed targets. I included all drugs, though, but pretty much all of the highest-enriched drugs are have a broad range of protein targets

### CODE REVIEW

File: 30-viper-drug-enrichment-bydrug.Rmd [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+```{r message=FALSE, warning=FALSE}
+
+term2gene <- drug_targets %>% 
+  mutate(term = std_name) %>% 
+  select(term, gene) %>% 
+  distinct()
+
+tidy <- viper_res %>% 
+  group_by(latent_var) %>% 
+  filter(any(gene %in% term2gene$gene)) %>% ##remove LVs that have no overlap if any exist
```
Comment (sgosline) [inline 30-viper-drug-enrichment-bydrug.Rmd]: Does GSEA fail if this is not filtered? This means that you're only looking at those genes/proteins that are drug targets, which seems off.

Comment (allaway) [reply 30-viper-drug-enrichment-bydrug.Rmd]: The `any()` makes it so it only filters out LV-viper protein groups if there are no druggable targets in each group. At least, I think that's what I was going for there. Either way, this `filter` has no effect, removing yields the same dataset.

**Comment by allaway** [comment]
Whoops, didn't realize it was in the wrong directory. just fixed that!

File: 30-viper-drug-enrichment-bydrug.Rmd (hunk 1 above)
Comment (allaway) [reply 30-viper-drug-enrichment-bydrug.Rmd]: I removed this because I'm not 100% sure it works the way I think it would...

Comment (allaway) [reply 30-viper-drug-enrichment-bydrug.Rmd]: Actually, it does work that way: https://stackoverflow.com/questions/40825037/filter-groups-by-occurrence-of-a-value

Comment (allaway) [reply 30-viper-drug-enrichment-bydrug.Rmd]: But either way it makes no difference to the result :)

**Comment by sgosline** [comment]
Re-cap of offline conversation:  I think it'd be helpful to include bargraph of number of targets for each drug alongside the enrichment plots

Additional question - is the foldchange in the network actually correlation? If so this is interesting.

File: 30-viper-drug-enrichment-bydrug.Rmd [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  if(nrow(x@result)==0){
+    TRUE
+  }else{
+    FALSE
+  }
+})
+
+##remove if no results
+res_complt <- res[empty_idx==F]
+
```
Comment (sgosline) [inline 30-viper-drug-enrichment-bydrug.Rmd]: Can you post a table of the the drug x LV enrichment scores and p-values to Synapse? I'd like to pull it down in the correlation analysis.

**Comment by allaway** [comment]
the latest version of this pr has this code:
```

tab <- lapply(names(res_complt), function(x){
  foo <- res_complt[[x]]@result %>%
    mutate(latent_var = x) %>%
    select(latent_var, everything())
}) %>% bind_rows()

# lv_drug_tab <- synBuildTable("Latent Variable Drug Set Enrichment
Analysis", "syn21046734", tab)
# synStore(lv_drug_tab)

DT::datatable(tab)
```
pushed to this table:
https://www.synapse.org/#!Synapse:syn21442304/tables/

**Comment by allaway** [comment]
> Re-cap of offline conversation: I think it'd be helpful to include bargraph of number of targets for each drug alongside the enrichment plots
> 
> Additional question - is the foldchange in the network actually correlation? If so this is interesting.

@sgosline barplots are added in the latest version of this branch!

**Pull request merged**